### PR TITLE
test: add eth.test.ts

### PR DESCRIPTION
**Pull request by keroro520**
In this PR, I have rewritten [`packages/api-server/tests/methods/eth.test.js`](./packages/api-server/tests/methods/eth.test.js) into TypeScript version [`packages/api-server/tests/methods/eth.test.ts`](./packages/api-server/tests/methods/eth.test.ts). There is no special reason, just because I don't know how to run the original JavaScript test cases properly. So I try to rewrite it in TypeScript. And I don't know why the `jayson` HTTP client isn't working in Ava, so I'm using `jayson/promise` instead.

### CODE REVIEW

File: packages/api-server/tests/www.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const jayson = require("jayson/promise");
```
Comment (classicalliu) [inline packages/api-server/tests/www.ts:1]:
```suggestion
import jayson from "jayson/promise";
```

File: packages/api-server/tests/www.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+export const client = jayson.Client.http({
```
Comment (classicalliu) [inline packages/api-server/tests/www.ts:3]: Why not use `RPC` as project used?

File: packages/api-server/tests/methods/eth.test.ts [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    t.pass();
+    return;
+  }
+
+  const res: JSONResponse = await client.request(t.title, [
+    block.number,
+    "0x0",
+  ]);
+  t.falsy(res.error);
+  t.is(JSON.stringify(res.result), JSON.stringify(block.transactions[0]));
```
Comment (classicalliu) [inline packages/api-server/tests/methods/eth.test.ts:63]:
```suggestion
  t.deepEqual(res.result, block.transactions[0]);
```

File: packages/api-server/tests/methods/eth.test.ts [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+test("eth_getTransactionByBlockHashAndIndex", async (t) => {
+  const block = t.context as EthBlock;
+  if (block.transactions.length === 0) {
+    t.pass();
+    return;
+  }
+
+  const res: JSONResponse = await client.request(t.title, [block.hash, "0x0"]);
+  t.falsy(res.error);
+  t.is(JSON.stringify(res.result), JSON.stringify(block.transactions[0]));
```
Comment (classicalliu) [inline packages/api-server/tests/methods/eth.test.ts:75]:
```suggestion
  t.deepEqual(res.result, block.transactions[0]);
```

File: packages/api-server/tests/methods/eth.test.ts [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+  if (block.transactions.length === 0) {
+    t.pass();
+    return;
+  }
+
+  const res: JSONResponse = await client.request(t.title, [
+    (block.transactions[0] as EthTransaction).hash,
+  ]);
+  t.falsy(res.error);
+  t.is(JSON.stringify(res.result), JSON.stringify(block.transactions[0]));
```
Comment (classicalliu) [inline packages/api-server/tests/methods/eth.test.ts:89]:
```suggestion
  t.deepEqual(res.result, block.transactions[0]);
```

File: packages/api-server/tests/methods/eth.test.ts [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+test("eth_hashrate", async (t) => {
+  const res: JSONResponse = await client.request(t.title, []);
+  t.falsy(res.error);
+  t.is(res.result, "0x0");
+});
+
+test("eth_accounts", async (t) => {
+  const res: JSONResponse = await client.request(t.title, []);
+  t.falsy(res.error);
+  t.is(JSON.stringify(res.result), JSON.stringify([]));
```
Comment (classicalliu) [inline packages/api-server/tests/methods/eth.test.ts:124]:
```suggestion
  t.deepEqual(res.result, []);
```

File: packages/api-server/tests/methods/eth.test.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+
+    let block: EthBlock = res.result!;
+    if (block.transactions.length > 0) {
+      return block;
+    }
+
+    number++;
+  }
+
+  console.warn("There is not non-empty block.");
```
Comment (classicalliu) [inline packages/api-server/tests/methods/eth.test.ts:151]:
```suggestion
  console.warn("There is no non-empty block.");
```

File: packages/api-server/tests/methods/net.test.ts [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+test("net_peerCount", async (t) => {
+  let res: JSONResponse = await client.request(t.title, []);
+  t.falsy(res.error);
+  t.is(res.result, "0x0");
+});
+
+test("net_listening", async (t) => {
+  let res: JSONResponse = await client.request(t.title, []);
+  t.falsy(res.error);
+  t.is(res.result, true);
```
Comment (classicalliu) [inline packages/api-server/tests/methods/net.test.ts:18]:
```suggestion
  t.true(res.result);
```

File: packages/api-server/tests/www.ts (hunk 2 above)
Comment (keroro520) [reply packages/api-server/tests/www.ts:3]: Do you mean [`godwoken::RPC`](https://github.com/nervosnetwork/godwoken-web3/blob/main/packages/godwoken/src/rpc.ts#L12-L23)?
I just followed [the previous code](https://github.com/nervosnetwork/godwoken-web3/pull/254/files#diff-d76812ac178d35280a797572cf21c8d2736bd4be013cb1c5341a1fc3b6fea4fcL8-L10). Do you think it's better to use [`godwoken::RPC`](https://github.com/nervosnetwork/godwoken-web3/blob/main/packages/godwoken/src/rpc.ts#L12-L23)? If so, we can do it in another PR.

**Comment by keroro520** [comment]
@classicalliu Thanks for your review comments and I have applied all resolved comments in the newest commit. PTAL.

File: packages/api-server/tests/www.ts (hunk 2 above)
Comment (classicalliu) [reply packages/api-server/tests/www.ts:3]: I mean `import { RPC } from "@ckb-lumos/toolkit"`, I just want to know why you use jayson client instead

**Review by classicalliu — APPROVED** [review]

**Pull request merged**